core/server/master/src/main/java/alluxio/master/file/meta/cross/cluster/CrossClusterMountClientRunner.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta.cross.cluster;
+
+import alluxio.client.cross.cluster.CrossClusterClient;
+import alluxio.collections.Pair;
+import alluxio.exception.runtime.AlluxioRuntimeException;
+import alluxio.exception.status.AlluxioStatusException;
+import alluxio.exception.status.UnavailableException;
+import alluxio.grpc.ErrorType;
+import alluxio.proto.journal.CrossCluster.MountList;
+import alluxio.resource.LockResource;
+
+import io.grpc.Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Runs a thread that will keep the cross cluster configuration service
+ * {@link alluxio.master.cross.cluster.CrossClusterState} up to date when local mount lists change.
+ * This runs a thread that continues to update the configuration service in case of
+ * failures.
+ */
+public class CrossClusterMountClientRunner implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(CrossClusterMountClientRunner.class);
+
+  private CrossClusterClient mRunnerClient;
+  private CrossClusterClient mMountChangeClient;
+  private final AtomicReference<Pair<Boolean, MountList>> mMountList =
+      new AtomicReference<>(new Pair<>(true, null));
+  private Thread mRunner;
+  private volatile boolean mDone = false;
+  private volatile boolean mStopped = true;
+  private volatile boolean mThreadChange = false;
+  private final Lock mClientChangeLock = new ReentrantLock();
+
+  /**
+   * Create a new cross cluster runner that will keep the configuration service up to date
```

Comment:
This thread is needed when the cross cluster master restarts, it will be used to keep the new master up to date